dm/dump_worker.go

```diff
@@ -0,0 +1,72 @@
+package dm
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/tiflow/dm/dm/unit"
+	"github.com/pingcap/tiflow/dm/dumpling"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+
+	"github.com/hanfei1991/microcosm/lib"
+	"github.com/hanfei1991/microcosm/model"
+)
+
+var _ lib.WorkerImpl = &dumpWorker{}
+
+type dumpWorker struct {
+	*lib.DefaultBaseWorker
+
+	cfg        *config.SubTaskConfig
+	unitHolder *unitHolder
+}
+
+func newDumpWorker(cfg lib.WorkerConfig) lib.Worker {
+	subtaskCfg := cfg.(*config.SubTaskConfig)
+	return &dumpWorker{
+		cfg: subtaskCfg,
+	}
+}
+
+func (d *dumpWorker) InitImpl(ctx context.Context) error {
+	d.unitHolder = newUnitHolder(dumpling.NewDumpling(d.cfg))
+	return errors.Trace(d.unitHolder.init(ctx))
+}
+
+func (d *dumpWorker) Tick(ctx context.Context) error {
+	d.unitHolder.lazyProcess()
+
+	return nil
+}
+
+func (d *dumpWorker) Status() lib.WorkerStatus {
+	hasResult, result := d.unitHolder.getResult()
+	if !hasResult {
+		return lib.WorkerStatus{Code: lib.WorkerStatusNormal}
+	}
+	if len(result.Errors) > 0 {
+		return lib.WorkerStatus{
+			Code:         lib.WorkerStatusError,
+			ErrorMessage: unit.JoinProcessErrors(result.Errors),
+		}
+	}
+	// should I keep the status unchanged for second call, or runtime will not
```

Comment:
The API will be changed to `UpdateStatus()` so this problem will not be relevant.